### PR TITLE
enriches postgres revisions with txID and timestamp

### PR DESCRIPTION
**Pull request by vroldanbet**
consumers of the Datastore Watch API may want to access the postgres transaction of a revision, and
the recorded timestamp.

Since we don't want to load it in every call sites for performance reasons, a revision may be in a partial state, thus it should be retrieved via the accessor methods.

### CODE REVIEW

File: internal/datastore/postgres/revisions.go [hunk 1]
```diff
@@ -284,6 +288,22 @@ func (pr postgresRevision) mustMarshalBinary() []byte {
 	return serialized
 }
 
+func (pr postgresRevision) OptionalTransactionID() (xid8, bool) {
```
Comment (josephschorr) [inline internal/datastore/postgres/revisions.go:291]: Doc comment

File: internal/datastore/postgres/revisions.go [hunk 2]
```diff
@@ -284,6 +288,22 @@ func (pr postgresRevision) mustMarshalBinary() []byte {
...
+	if !pr.optionalTxID.Valid {
+		return xid8{}, false
+	}
+
+	return pr.optionalTxID, true
+}
+
+func (pr postgresRevision) OptionalTimestamp() (uint64, bool) {
```
Comment (josephschorr) [inline internal/datastore/postgres/revisions.go:299]: Doc comment

File: internal/datastore/postgres/revisions_test.go [hunk 3]
```diff
@@ -83,6 +84,23 @@ func TestRevisionSerDe(t *testing.T) {
...
+	require.True(t, ok)
+	require.Equal(t, receivedTimestamp, testTimestamp)
+	txid, ok := pgr.OptionalTransactionID()
+	require.True(t, ok)
+	require.Equal(t, newXid8(1), txid)
+
+	pgr = postgresRevision{snapshot: pgSnapshot{}}
+	_, ok = pgr.OptionalTimestamp()
+	require.False(t, ok)
+	_, ok = pgr.OptionalTransactionID()
```
Comment (josephschorr) [inline internal/datastore/postgres/revisions_test.go:100]: add tests for when it is available

File: pkg/datastore/test/counters.go [hunk 4]
```diff
@@ -174,7 +174,7 @@ func UpdateRelationshipCounterTest(t *testing.T, tester DatastoreTester) {
 	require.Len(t, filters, 1)
 	require.Equal(t, "somedocfilter", filters[0].Name)
 	require.Equal(t, 1234, filters[0].Count)
-	require.Equal(t, updatedRev, filters[0].ComputedAtRevision)
+	require.True(t, updatedRev.Equal(filters[0].ComputedAtRevision))
```
Comment (josephschorr) [inline pkg/datastore/test/counters.go:177]: Add a comment here as to why Equal is not used

Comment (vroldanbet) [reply pkg/datastore/test/counters.go:179]: https://github.com/authzed/spicedb/pull/1951/commits/dc450332fa3d03ed97d176a50d1e33686e9fdf13

File: internal/datastore/postgres/revisions_test.go [hunk 5]
```diff
@@ -83,6 +84,23 @@ func TestRevisionSerDe(t *testing.T) {
...
+	require.False(t, ok)
```
Comment (josephschorr) [inline internal/datastore/postgres/revisions_test.go:101]: Add an assertion that they are still equal here, so long as the snapshot is the same

Comment (vroldanbet) [reply internal/datastore/postgres/revisions_test.go:102]: done

**Pull request merged**